File: app/src/main/java/com/zibete/proyecto1/ReportActivity.java
package com.zibete.proyecto1;

import android.content.Intent;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import com.google.android.material.button.MaterialButton;
import com.google.android.material.dialog.MaterialAlertDialogBuilder;
import com.google.android.material.textfield.TextInputEditText;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.zibete.proyecto1.Splash.SplashActivity;
import com.zibete.proyecto1.utils.UserRepository;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Locale;

import static com.zibete.proyecto1.utils.FirebaseRefs.ref_zibe;

public class ReportActivity extends AppCompatActivity {

    private Toolbar toolbar_ajustes;
    private TextInputEditText edt_comentarios;
    private MaterialButton btn_send;
    private @Nullable FirebaseUser user;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.dialog_report);

        // Toolbar
        toolbar_ajustes = findViewById(R.id.toolbar_ajustes);
        setSupportActionBar(toolbar_ajustes);
        if (getSupportActionBar() != null) {
            getSupportActionBar().setTitle("");
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        }
        toolbar_ajustes.setNavigationOnClickListener(v -> getOnBackPressedDispatcher().onBackPressed());

        // Views
        edt_comentarios = findViewById(R.id.edt_comentarios);
        btn_send = findViewById(R.id.btn_send);

        // Estado inicial
        btn_send.setEnabled(false);
        user = FirebaseAuth.getInstance().getCurrentUser();

        // Habilitar botón si hay texto
        edt_comentarios.addTextChangedListener(new TextWatcher() {
            @Override public void beforeTextChanged(CharSequence s, int start, int count, int after) { }
            @Override public void onTextChanged(CharSequence s, int start, int before, int count) {
                btn_send.setEnabled(s != null && s.toString().trim().length() > 0);
            }
            @Override public void afterTextChanged(Editable s) {
                btn_send.setEnabled(s != null && s.toString().trim().length() > 0);
            }
        });

        // Enviar
        btn_send.setOnClickListener(v -> {
            String mensaje = edt_comentarios.getText() == null ? "" : edt_comentarios.getText().toString().trim();
            if (mensaje.isEmpty()) {
                btn_send.setEnabled(false);
                return;
            }

            // Timestamp legible
            String date = new SimpleDateFormat("yyyy/MM/dd/HH:mm:ss", Locale.getDefault())
                    .format(Calendar.getInstance().getTime());

            if (user != null) {
                ref_zibe.child("Comentarios").child(date).child("ID").setValue(user.getUid());
                ref_zibe.child("Comentarios").child(date).child("nombre").setValue(user.getDisplayName());
                ref_zibe.child("Comentarios").child(date).child("email").setValue(user.getEmail());
            }
            ref_zibe.child("Comentarios").child(date).child("mensaje").setValue(mensaje);

            // Dialogo Material con overlay Zibe.Dialog (definido en styles)
            new MaterialAlertDialogBuilder(ReportActivity.this)
                    .setTitle("¡Mensaje enviado!")
                    .setMessage("¡Muchas gracias! El mensaje ha sido enviado a nuestro equipo de soporte.")
                    .setCancelable(false)
                    .setPositiveButton("OK", (dlg, idBtn) -> {
                        startActivity(new Intent(ReportActivity.this, SplashActivity.class));
                        finish();
                    })
                    .show();
        });
    }

    @Override
    protected void onPause() {
        super.onPause();
        if (user != null) UserRepository.setUserOffline(getApplicationContext(),user.getUid());
    }

    @Override
    protected void onResume() {
        super.onResume();
        if (user != null) UserRepository.setUserOnline(getApplicationContext(), user.getUid());
    }
}

File: app/src/main/java/com/zibete/proyecto1/ReportActivity.kt
package com.zibete.proyecto1

import android.content.DialogInterface
import android.content.Intent
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.appbar.MaterialToolbar
import com.google.android.material.button.MaterialButton
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import com.google.android.material.textfield.TextInputEditText
import com.google.firebase.auth.FirebaseAuth
import com.zibete.proyecto1.Splash.SplashActivity
import com.zibete.proyecto1.utils.FirebaseRefs.refZibe
import com.zibete.proyecto1.utils.UserRepository.setUserOffline
import com.zibete.proyecto1.utils.UserRepository.setUserOnline
import java.text.SimpleDateFormat
import java.util.Calendar
import java.util.Locale

class ReportActivity : AppCompatActivity() {

    private lateinit var toolbar: MaterialToolbar
    private lateinit var edtComentarios: TextInputEditText
    private lateinit var btnSend: MaterialButton

    private val user = FirebaseAuth.getInstance().currentUser

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.dialog_report)

        setupToolbar()
        setupViews()
        setupListeners()
    }

    private fun setupToolbar() {
        toolbar = findViewById(R.id.toolbar_ajustes) // Asegurate que en dialog_report sea MaterialToolbar
        setSupportActionBar(toolbar)

        supportActionBar?.apply {
            title = ""
            setDisplayHomeAsUpEnabled(true)
        }

        toolbar.setNavigationOnClickListener {
            onBackPressedDispatcher.onBackPressed()
        }
    }

    private fun setupViews() {
        edtComentarios = findViewById(R.id.edt_comentarios)
        btnSend = findViewById(R.id.btn_send)
        btnSend.isEnabled = false
    }

    private fun setupListeners() {
        // Habilitar botón solo si hay texto
        edtComentarios.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
                btnSend.isEnabled = !s.isNullOrBlank()
            }

            override fun afterTextChanged(s: Editable?) {
                btnSend.isEnabled = !s.isNullOrBlank()
            }
        })

        btnSend.setOnClickListener {
            val mensaje = edtComentarios.text?.toString()?.trim().orEmpty()
            if (mensaje.isEmpty()) {
                btnSend.isEnabled = false
                return@setOnClickListener
            }

            val dateKey = SimpleDateFormat("yyyy/MM/dd/HH:mm:ss", Locale.getDefault())
                .format(Calendar.getInstance().time)

            // Guarda datos del usuario si está logueado
            user?.let { u ->
                refZibe.child("Comentarios").child(dateKey).apply {
                    child("ID").setValue(u.uid)
                    child("nombre").setValue(u.displayName)
                    child("email").setValue(u.email)
                }
            }

            // Guarda mensaje
            refZibe.child("Comentarios").child(dateKey)
                .child("mensaje")
                .setValue(mensaje)

            // Dialog Material
            MaterialAlertDialogBuilder(this)
                .setTitle("¡Mensaje enviado!")
                .setMessage("¡Muchas gracias! El mensaje ha sido enviado a nuestro equipo de soporte.")
                .setCancelable(false)
                .setPositiveButton("OK") { _: DialogInterface, _: Int ->
                    startActivity(Intent(this, SplashActivity::class.java))
                    finish()
                }
                .show()
        }
    }

    override fun onPause() {
        super.onPause()
        user?.uid?.let { setUserOffline(applicationContext, it) }
    }

    override fun onResume() {
        super.onResume()
        user?.uid?.let { setUserOnline(applicationContext, it) }
    }
}
